### PR TITLE
Handles checksum variations. Creates label when missing.

### LINKED ISSUE #837: sha1 not mapping correctly
For https://argo-stage.stanford.edu/view/druid:hx013yf6680?beta=true, the sha1 is mapping incorrectly. For example:
```
"hasMessageDigests": [
                {
                  "type": "sha1",
                  "digest": ""
                },
                {
                  "type": "md5",
                  "digest": "86c3108e4fb762d2b705fe9e64a0d9a2"
                }
              ]
```

The contentMetadata has:
```
<file dataType="WARC" publish="no" shelve="yes" preserve="yes" id="ARCHIVEIT-5425-TEST-JOB842631-SEED1999838-20190424183402159-00000-h3.warc.gz" size="337857033" mimetype="application/octet-stream">
      <checksum type="MD5">86c3108e4fb762d2b705fe9e64a0d9a2</checksum>
      <checksum type="SHA1">ffed47da235144367c62f9afa09c7394de612136</checksum>
    </file>
  </resource>
```

### PR DESCRIPTION
**Pull request by justinlittman**
closes #837

## Why was this change made?
WAS is "special".

## Was the API documentation (openapi.yml) updated?
No.

## Does this change affect how this application integrates with other services?
If so, please confirm change was tested on stage and/or test added to sul-dlss/infrastructure-integration-test.
No.

### CODE REVIEW

File: app/services/cocina/dro_structural_builder.rb [hunk 1]
```diff
@@ -70,10 +69,23 @@ def build_files(file_nodes, version:, parent_id:)
...
+
+    # The old google books use upcased versions. See https://argo.stanford.edu/view/druid:dd116zh0343
+    def md5_from(node)
+      node.xpath('checksum[@type="md5"]').text.presence || node.xpath('checksum[@type="MD5"]').text.presence
+    end
+
+    # The old google books use upcased versions. See https://argo.stanford.edu/view/druid:dd116zh0343
+    # Web archive crawls use SHA1
+    def sha1_from(node)
+      node.xpath('checksum[@type="sha1"]').text.presence || node.xpath('checksum[@type="SHA1"]').text.presence || node.xpath('checksum[@type="SHA-1"]').text.presence
```
Comment (jcoyne) [inline app/services/cocina/dro_structural_builder.rb:88]: could you do `'checksum[(@type="sha1" or @type="SHA1" @type="SHA-1")]` as your xpath?

File: app/services/cocina/dro_structural_builder.rb [hunk 2]
```diff
@@ -70,10 +69,23 @@ def build_files(file_nodes, version:, parent_id:)
...
           attrs[:hasMessageDigests] << { type: 'sha1', digest: sha1 } if sha1
+          md5 = md5_from(node)
           attrs[:hasMessageDigests] << { type: 'md5', digest: md5 } if md5
         end
       end
     end
+
+    # The old google books use upcased versions. See https://argo.stanford.edu/view/druid:dd116zh0343
+    def md5_from(node)
+      node.xpath('checksum[@type="md5"]').text.presence || node.xpath('checksum[@type="MD5"]').text.presence
```
Comment (jcoyne) [inline app/services/cocina/dro_structural_builder.rb:82]: could you do 'checksum[(@type="md5" or @type="MD5")] as your xpath?

File: app/services/cocina/dro_structural_builder.rb (hunk 1 above)
Comment (justinlittman) [reply app/services/cocina/dro_structural_builder.rb]: Yes, much cleaner. Thanks.

File: app/services/cocina/dro_structural_builder.rb (hunk 2 above)
Comment (justinlittman) [reply app/services/cocina/dro_structural_builder.rb]: Done.

**Pull request merged**